File: 프로그래머스/0/120825. 문자 반복 출력하기/문자 반복 출력하기.java
class Solution {
    public String solution(String my_string, int n) {
        String answer = "";
        for(int i=0;i<my_string.length();i++){
            answer += Character.toString(my_string.charAt(i)).repeat(n);
        }
        return answer;
    }
}

// 다른 풀이
// 2중 for문을 활용한 풀이 
// 가장 많은 사람들의 풀이 방법이다.
class Solution {
    public String solution(String my_string, int n) {
        String answer = "";

        for(int i=0; i<my_string.length(); i++){
            for(int j=0; j<n; j++){
                answer+=my_string.charAt(i);
            }
        }

        return answer;
    }
}
// StringBuilder를 활용한 풀이
class Solution {
    public String solution(String my_string, int n) {
        StringBuilder sb = new StringBuilder(); // 객체 생성
        for(char c : my_string.toCharArray()){ // 문자열을 문자 배열로 변형한 것을 for-each문을 사용해 문자 변수 c에 담는다.
            sb.append((c + "").repeat(n)); // 문자 c를 문자열로 변형하여 n번 반복후 객체에 집어 넣는다.
        }
        return sb.toString(); StringBuilder // 객체의 내용을 String으로 변환하여 반환
    }
}
// split()을 사용해서 문자열 배열생성하는 방법
class Solution {
    public String solution(String my_string, int n) {
        String answer = "";
        String[] str = my_string.split(""); // 입력받은 문자열을 한 글자씩 문자열로 쪼개서 배열로 만든다.
        for(int i=0; i<my_string.length(); i++){
            answer += str[i].repeat(n); // repeat를 사용하여 반복
        }
        return answer;
    }
}
// 문자열 연결 시, 기존 문자열이 불변(immutable)이므로, 새로운 문자열 객체가 매번 생성되서 성능 저하가 예상된다.
// StringBuilder를 사용하는 것이 좀 더 효율적일듯
// 하지만 가독성은 이게 더 좋은듯하다.

File: 프로그래머스/0/120825. 문자 반복 출력하기/문자 반복 출력하기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

// 파라미터로 주어지는 문자열은 const로 주어집니다. 변경하려면 문자열을 복사해서 사용하세요.
char* solution(const char* my_string, int n) {
    int size = strlen(my_string);
    // return 값은 malloc 등 동적 할당을 사용해주세요. 할당 길이는 상황에 맞게 변경해주세요.
    char* answer = (char*)malloc((size * n) + 1);
    int pointer = 0;
    for(int idx = 0; idx < size; idx++)
    {
        for(int i = 0; i < n; i++)
        {
            answer[pointer++] = my_string[idx];
        }
    }
    answer[pointer] = '\0';
    return answer;
}